SonarQube.MSBuild.Tasks/WriteProjectInfoFile.cs

```diff
@@ -60,12 +74,12 @@ public override bool Execute()
             pi.ProjectName = this.ProjectName;
             pi.FullPath = this.FullProjectPath;
 
-            // TODO: handle failures and missing values.
             Guid projectId;
             if (Guid.TryParse(this.ProjectGuid, out projectId))
             {
                 pi.ProjectGuid = projectId;
                 pi.AnalysisResults = TryCreateAnalysisResults(this.AnalysisResults);
+                pi.AnalysisSettings = TryCreateAnalysisSettings(this.AnalysisSettings);
```

Comment:
I think there is an FxCop rule that checks it for public methods, but not private ones.

Personally I don't find it to be a particularly useful pattern and don't generally follow it for private code if the method can return null to indicate failure (which is the case here).